pkgs/development/libraries/gcr/default.nix

```diff
@@ -79,26 +70,35 @@ stdenv.mkDerivation rec {
   ];
 
   mesonFlags = [
-    "-Dgtk_doc=false"
     # We are still using ssh-agent from gnome-keyring.
     # https://github.com/NixOS/nixpkgs/issues/140824
     "-Dssh_agent=false"
+  ] ++ lib.optionals (!stdenv.isLinux) [
+    "-Dsystemd=disabled"
   ];
 
   doCheck = false; # fails 21 out of 603 tests, needs dbus daemon
 
   PKG_CONFIG_SYSTEMD_SYSTEMDUSERUNITDIR = "${placeholder "out"}/lib/systemd/user";
 
   postPatch = ''
-    patchShebangs build/ gcr/fixtures/
+    patchShebangs gcr/fixtures/
 
     chmod +x meson_post_install.py
     patchShebangs meson_post_install.py
+  '' + lib.optionalString stdenv.isDarwin ''
+    substituteInPlace meson_post_install.py --replace ".so" "${stdenv.hostPlatform.extensions.sharedLibrary}"
```

Comment:
This wasn't what @SuperSandro2000 was suggesting in https://github.com/NixOS/nixpkgs/pull/185707#discussion_r949522100. The `isDarwin` check is not needed.

Reply:
Fixed, thanks.